Bottom: 431fe76cbf6de6ef7093ecc7973016feca6529ac
Top:    dcf215f43282ba193897c00d3d5dec159f5f6a37
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-15 12:03:35 +0800

Refresh of virtio-net

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 3ce5f37..f573614 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -356,10 +356,9 @@ static void virtnet_rx_recycle(struct receive_queue *rq, struct page *page)
 {
 	struct virtnet_page_cache *cache = &rq->page_cache;
 
-	if (page_ref_count(page) == 1 && cache->index < VIRTNET_CACHE_SIZE)
+	if (page_ref_count(page) == 2 && cache->index < VIRTNET_CACHE_SIZE)
 		cache->pages[cache->index++] = page;
-	else
-		put_page(page);
+	put_page(page);
 }
 
 static bool virtnet_xdp_xmit(struct virtnet_info *vi,
